File: scripts/simple_number_publisher.py
#!/usr/bin/env python3
"""
    ROS publisher node publishing an incrementing number 
    
    Author: Jari Honkanen
"""

import rospy
from std_msgs.msg import Int64

if __name__ == '__main__':

	rospy.init_node("simple_number_publisher")
	pub = rospy.Publisher("/number", Int64, queue_size=10)

	rospy.loginfo("[INFO] simple_number_publisher started")

	rate = rospy.Rate(0.2)    # Publish every 5 seconds

	number = 1

	while not rospy.is_shutdown():

		msg = Int64()
		msg.data = number 
		pub.publish(msg)
		rospy.loginfo("Published number " + str(number))
		number += 1
		rate.sleep()

File: my_simple_robot_nodes/src/simple_number_publisher.cpp
/*-------------------------------------------------------------------*\
  NAME
    simple_number_publisher.cpp

  DESCRIPTION
    ROS publisher node publishing an incrementing number 

  AUTHOR
    Jari Honkanen
\*-------------------------------------------------------------------*/

#include <ros/ros.h>
#include <std_msgs/Int64.h>

int main (int argc, char **argv)
{
	ros::init(argc, argv, "simple_number_publisher", ros::init_options::AnonymousName);
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::Int64>("/number", 10);
	ros::Rate rate(0.2);    // publish every 5 seconds

	ROS_INFO("[INFO] simple_number_publisher started");

	int number = 1;

	while (ros::ok()) {
		std_msgs::Int64 msg;
		msg.data = number;
		number++;
		pub.publish(msg);
		ROS_INFO("Published number %d", number);
		rate.sleep();
	}
}
